src/accelerate/accelerator.py

```diff
@@ -2716,10 +2725,16 @@ def _inner(folder):
             hook(self._models, weights, output_dir)
 
         save_location = save_accelerator_state(
-            output_dir, weights, optimizers, schedulers, self.state.process_index, self.scaler
+            output_dir,
+            weights,
+            optimizers,
+            schedulers,
+            self.state.process_index,
+            self.scaler,
+            self.project_configuration.save_on_each_node,
         )
         for i, obj in enumerate(self._custom_objects):
-            save_custom_state(obj, output_dir, i)
+            save_custom_state(obj, output_dir, i, self.project_configuration.save_on_each_node)
```

Comment:
Here and below, I'd personally prefer to pass `save_on_each_node` as a keyword argument.

Reply:
Done